### PR TITLE
feat(ListComposition): filter on limited column list

### PR DESCRIPTION
**Pull request by frassinier**
**What is the problem this PR is trying to solve?**
By default, we perform a filter on each property of the resource we're currently listing

**What is the chosen solution to this problem?**
Be able to precise on which columns we accept to filter using `<TextFilter applyOn={[ 'some', 'columns', 'keys' ]} />`

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

https://jira.talendforge.org/browse/TCCUI-201

### CODE REVIEW

File: packages/components/src/List/ListComposition.stories.js [hunk 1]
```diff
@@ -246,7 +246,7 @@ storiesOf('Data/List/List Composition', module)
 >
 	<List.Toolbar>
 		<List.Toolbar.Right>
-			<List.TextFilter id="my-list-textFilter" />
+			<List.TextFilter id="my-list-textFilter" applyOn={['name', 'description']} />
```
Comment (ncomont) [inline packages/components/src/List/ListComposition.stories.js:249]: Could you add a specific story for that ? Or ar least a note in the story description ?

File: packages/components/src/List/ListComposition/Manager/hooks/useCollectionFilter.hook.js [hunk 2]
```diff
@@ -22,14 +22,17 @@ function defaultFilterFunction(value, textFilter) {
...
 	if (!textFilter) {
 		return collection;
 	}
 
 	const normalizedTextFilter = normalizeInput(textFilter);
 	return collection.filter(item =>
 		Object.entries(item).find(([key, value]) => {
+			if (filteredColumns && Array.isArray(filteredColumns) && !filteredColumns.includes(key)) {
+				return false;
+			}
```
Comment (jmfrancois) [inline packages/components/src/List/ListComposition/Manager/hooks/useCollectionFilter.hook.js:35]: you just have to not have the filterFunction for those key !

File: packages/components/src/List/ListComposition/Manager/hooks/useCollectionFilter.hook.test.js [hunk 3]
```diff
@@ -139,6 +150,27 @@ describe('useCollectionFilter', () => {
...
+		const wrapper = mount(
+			<FilterComponent collection={collection} initialFilteredColumns={['lastName']} />,
+		);
+
+		let filteredCollection = wrapper.find('#mainChild').prop('filteredCollection');
+		expect(filteredCollection).toBe(collection);
+
+		// when
+		act(() => {
+			wrapper.find('#mainChild').prop('setTextFilter')('l');
```
Comment (ncomont) [inline packages/components/src/List/ListComposition/Manager/hooks/useCollectionFilter.hook.test.js:164]: I can be wrong, but for me this test does not test anything. `Silva Madder` is the first item with an `l` in its row.

File: packages/components/src/List/ListComposition/Manager/hooks/useCollectionFilter.hook.test.js [hunk 4]
```diff
@@ -139,6 +150,27 @@ describe('useCollectionFilter', () => {
...
+		});
+		wrapper.update();
+
+		// then
+		filteredCollection = wrapper.find('#mainChild').prop('filteredCollection');
+		expect(filteredCollection).toHaveLength(1);
```
Comment (ncomont) [inline packages/components/src/List/ListComposition/Manager/hooks/useCollectionFilter.hook.test.js:170]: ok forgive me 🙃

**Review by ncomont — APPROVED** [review]

File: packages/components/src/List/ListComposition/TextFilter/TextFilter.component.test.js [hunk 5]
```diff
@@ -66,6 +67,18 @@ describe('TextFilter', () => {
...
+	it('should deal with columns on which apply filter', () => {
+		// when
+		mount(
+			<ListContext.Provider value={defaultContext}>
+				<TextFilter id="myTextFilter" applyOn={['foo']} />
+			</ListContext.Provider>,
+		);
+
+		// then
+		expect(defaultContext.setFilteredColumns).toHaveBeenCalledWith(['foo']);
```
Comment (jmfrancois) [inline packages/components/src/List/ListComposition/TextFilter/TextFilter.component.test.js:79]: defaultContext is not reset in a before each !

File: packages/components/src/List/ListComposition/TextFilter/TextFilter.component.test.js [hunk 6]
```diff
@@ -12,6 +12,7 @@ describe('TextFilter', () => {
 	const defaultContext = {
 		textFilter: '',
 		setTextFilter: jest.fn(),
+		setFilteredColumns: jest.fn(),
```
Comment (jmfrancois) [inline packages/components/src/List/ListComposition/TextFilter/TextFilter.component.test.js:15]: it should be 
```
let defaultContext
beforeEach(() => {
	const defaultContext = {
		textFilter: '',
		textFilter: '',
		setTextFilter: jest.fn(),
		setTextFilter: jest.fn(),
		setFilteredColumns: jest.fn(),
		t: getDefaultT(),
		t: getDefaultT(),
	};
	};
```

File: packages/components/src/List/ListComposition/Manager/ListManager.component.test.js [hunk 7]
```diff
@@ -91,10 +96,42 @@ describe('List Manager', () => {
...
+		act(() => {
+			const setFilteredColumns = wrapper.find(TestConsumer).prop('setFilteredColumns');
+			setFilteredColumns(filteredColumns);
+		});
+		wrapper.update();
+
+		// then
+		expect(wrapper.find(TestConsumer).prop('filteredColumns')).toBe(filteredColumns);
+	});
+
```
Comment (jsomsanith-tlnd) [inline packages/components/src/List/ListComposition/Manager/ListManager.component.test.js:125]: Please don't do this kind of tests, that test an internal implementation detail.

**Review by jsomsanith-tlnd — APPROVED** [review]

**Pull request merged**